src/languages/yaml.js

```diff
@@ -20,11 +20,16 @@ export default function(hljs) {
   const KEY = {
     className: 'attr',
     variants: [
-      { begin: '\\w[\\w :\\/.-]*:(?=[ \t]|$)' },
+      // added brackets support 
+      { begin: '\\w[\\w :\\(\\)\\/.-]*:(?=[ \t]|$)' },
       { // double quoted keys
         begin: '"\\w[\\w :\\/.-]*":(?=[ \t]|$)' },
       { // single quoted keys
-        begin: '\'\\w[\\w :\\/.-]*\':(?=[ \t]|$)' }
+        begin: '\'\\w[\\w :\\/.-]*\':(?=[ \t]|$)' },
+      { // double quoted keys - with brackets
+        begin: '"\\w[\\w :\\(\\)\\/.-]*":(?=[ \t]|$)' },
+      { // single quoted keys - with brackets
```

Comment:
Now that we have these, do we need the "without brackets" variants, doesn't this cover both cases?

Reply:
Hi @joshgoebel , My bad sorry :sweat_smile: . Have changed it

Reply:
Nice.  Now can we convert them to proper regex with `//` not strings and simplify all that escaping?  I'm pretty sure the `\t` escaping it already broken since it's not double escaped....

Reply:
Hey @joshgoebel

Just as a matter of interest, I checked if \t is broken or not and the \t matching is working fine I believe, even without double escaping. I added a key and put a tab after the colon and it was highlighted correctly. (Check the last key below). Let me know if I tested it correctly :sweat_smile: 
![image](https://github.com/highlightjs/highlight.js/assets/37235937/47172fd6-726b-42fb-a49a-f227e2b4ae5f)

Also I have made the changes suggested.

Just out of curiosity I noticed there are several other places also where instead of JavaScript RegExp objects we are using Strings to represent regex.
for example
https://github.com/highlightjs/highlight.js/blob/17494c81dbade12834feedddf6fb208bcc9cbcce/src/languages/yaml.js#L78-L81

Is there any reason to prefer strings over regexp object? Just wanted to get a better understanding of the project. 
Thanks in advance.

Reply:
> Is there any reason to prefer strings over regexp object? 

Sometimes when we build larger regex with `+` you need strings... but we have `regex.join` also that can take regex... so before just changing everything you have to look at everywhere it's used to make sure you aren't breaking anything.

Reply:
>  I checked if \t is broken or not and the \t matching is working fine I believe, even without double escaping

Right it would, but in one case you have a _literal_ tab and the other you have a \t escape sequence... both will match a tab of course, hence the same behavior.

Reply:
Alright! I got it. Thanks

Reply:
Is there anything else that needs to be done?